ee/clickhouse/views/test/test_clickhouse_person.py

```diff
@@ -13,14 +15,38 @@ def _create_event(**kwargs):
 
 
 def _get_events():
-    return sync_execute("select * from events")
+    return sync_execute("SELECT * FROM events")
 
 
 def _create_person(**kwargs):
-    return Person.objects.create(**kwargs)
+    postgres_person: Person = Person.objects.create(**kwargs)
+    sync_execute(
```

Comment:
This should not be in here and will result in duplicate data.

There are already hooks defined in ee/clickhouse/models/person.py which create the user

In general trying to do inserts into clickhouse from python code is a failing strategy we should avoid - they will cause bugs as the python and plugin server behavior diverge. Example: https://github.com/PostHog/posthog/issues/5145

Reply:
I agree, though persons do need to be in the DB at the time of this test suite running, so not sure what cleaner solution there is. Sending events to the plugin server to create the persons? That's really asynchronous though, we'd have to `sleep` until the persons appear in CH here (which we do routinely do in plugin server tests, so maybe?).

Reply:
Clickhouse inserts are synchronous in tests, check out ee/kafka_client/client.py#ClickhouseProducer implementation